From 8d3c64c7f075241f228d6f16ea4d1baf6df15f4c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo@beauzee.fr>
Date: Fri, 24 Jun 2016 13:41:23 +0200
Subject: [PATCH] demux: Don't assert the stream isn't NULL when destroying it

---
 src/input/demux.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/input/demux.c b/src/input/demux.c
index 448d2ea..468efbd 100644
--- a/src/input/demux.c
+++ b/src/input/demux.c
@@ -95,7 +95,8 @@ typedef struct demux_priv_t
 
 static void demux_DestroyDemux(demux_t *demux)
 {
-    assert(demux->s != NULL);
+    if (demux->s == NULL)
+        return;
     stream_Delete(demux->s);
 }
 
-- 
2.8.1

