From 331255171c98db39e65e667510fe18088a3e5805 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo@beauzee.fr>
Date: Tue, 10 Feb 2015 11:58:54 +0100
Subject: [PATCH] wasapi: Don't fail if the client is already initialized

---
 modules/audio_output/wasapi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/audio_output/wasapi.c b/modules/audio_output/wasapi.c
index 63d0e91..3964ef3 100644
--- a/modules/audio_output/wasapi.c
+++ b/modules/audio_output/wasapi.c
@@ -447,7 +447,7 @@ static HRESULT Start(aout_stream_t *s, audio_sample_format_t *restrict fmt,
                                  AOUT_MAX_PREPARE_TIME * 10, 0,
                                  (hr == S_OK) ? &wf.Format : pwf, sid);
     CoTaskMemFree(pwf);
-    if (FAILED(hr))
+    if (FAILED(hr) && hr != AUDCLNT_E_ALREADY_INITIALIZED)
     {
         msg_Err(s, "cannot initialize audio client (error 0x%lx)", hr);
         goto error;
-- 
2.8.1

