From 6bf35ea010195f84e7ceb0a4d3a51cefb1d09b69 Mon Sep 17 00:00:00 2001
From: Jean-Baptiste Kempf <jb@videolan.org>
Date: Wed, 16 Apr 2014 16:09:53 +0200
Subject: [PATCH 3/9] Cheat for Windows Store subtitles

---
 lib/video.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/video.c b/lib/video.c
index c41c6e35ff..b001af07c0 100644
--- a/lib/video.c
+++ b/lib/video.c
@@ -407,7 +407,11 @@ int libvlc_video_set_subtitle_file( libvlc_media_player_t *p_mi,
         if( psz_mrl )
         {
             if( !input_AddSlave( p_input_thread, SLAVE_TYPE_SPU, psz_mrl,
+#if VLC_WINSTORE_APP
+                                 false, false ) )
+#else
                                  true, false ) )
+#endif
                 b_ret = true;
             free( psz_mrl );
         }
-- 
2.11.0

