From 1753fcf92174f49b84d83050d1825179c48e4171 Mon Sep 17 00:00:00 2001
From: Jean-Baptiste Kempf <jb@videolan.org>
Date: Wed, 16 Apr 2014 16:09:53 +0200
Subject: [PATCH 04/17] Cheat for Windows Store subtitles

---
 lib/video.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/video.c b/lib/video.c
index 8655a32..3e1389d 100644
--- a/lib/video.c
+++ b/lib/video.c
@@ -351,7 +351,11 @@ int libvlc_video_set_subtitle_file( libvlc_media_player_t *p_mi,
 
     if( p_input_thread )
     {
+#if VLC_WINSTORE_APP
+        if( !input_AddSubtitle( p_input_thread, psz_subtitle, false ) )
+#else
         if( !input_AddSubtitle( p_input_thread, psz_subtitle, true ) )
+#endif
             b_ret = true;
         vlc_object_release( p_input_thread );
     }
-- 
2.9.3

