From 415536ed6e20d447833ac83469e20361c8dc63d0 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@videolabs.io>
Date: Mon, 23 Jan 2017 13:11:22 +0100
Subject: [PATCH 11/55] vout: win32: name the parameters for more clarity

---
 modules/video_output/win32/common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/video_output/win32/common.h b/modules/video_output/win32/common.h
index e88184d..5d73bca 100644
--- a/modules/video_output/win32/common.h
+++ b/modules/video_output/win32/common.h
@@ -275,7 +275,7 @@ void CommonClean(vout_display_t *);
 void CommonManage(vout_display_t *);
 int  CommonControl(vout_display_t *, int , va_list );
 void CommonDisplay(vout_display_t *);
-int  CommonUpdatePicture(picture_t *, picture_t **, uint8_t *, unsigned);
+int  CommonUpdatePicture(picture_t *, picture_t **fallback, uint8_t *plane, unsigned pitch);
 
 void UpdateRects (vout_display_t *,
                   const vout_display_cfg_t *,
-- 
2.10.1.windows.1

