From 61a3ea34ef08c3fe23d0709e1c43a5c3b28c1b79 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@videolabs.io>
Date: Wed, 19 Oct 2016 18:15:50 +0200
Subject: [PATCH] d3d11va: FIXME workaround to avoid some files decoding
 nothing

---
 modules/codec/avcodec/d3d11va.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/modules/codec/avcodec/d3d11va.c b/modules/codec/avcodec/d3d11va.c
index e7ffbb0..6a430d3 100644
--- a/modules/codec/avcodec/d3d11va.c
+++ b/modules/codec/avcodec/d3d11va.c
@@ -968,6 +968,8 @@ static int DxCreateDecoderSurfaces(vlc_va_t *va, int codec_id, const video_forma
     viewDesc.DecodeProfile = dx_sys->input;
     viewDesc.ViewDimension = D3D11_VDOV_DIMENSION_TEXTURE2D;
 
+sys->b_extern_pool=false;
+
     if (sys->b_extern_pool)
     {
         size_t surface_idx;
@@ -994,7 +996,7 @@ static int DxCreateDecoderSurfaces(vlc_va_t *va, int codec_id, const video_forma
             assert(texDesc.Format == sys->render);
             assert(texDesc.BindFlags & D3D11_BIND_DECODER);
 
-#if !LIBAVCODEC_VERSION_CHECK( 57, 27, 2, 61, 102 )
+#if false && !LIBAVCODEC_VERSION_CHECK( 57, 27, 2, 61, 102 )
             if (pic->p_sys->slice_index != surface_idx)
             {
                 msg_Warn(va, "d3d11va requires decoding slices to be the first in the texture (%d/%d)",
-- 
2.9.3

