From b7ac6cb44a0ec6e99f236f361b87ef7fbc0c3fa1 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@videolabs.io>
Date: Thu, 26 Jan 2017 10:31:44 +0100
Subject: [PATCH 12/55] direct3d11: warn when the projection is not handled

---
 modules/video_output/win32/direct3d11.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/modules/video_output/win32/direct3d11.c b/modules/video_output/win32/direct3d11.c
index 24f566d..622bb2a 100644
--- a/modules/video_output/win32/direct3d11.c
+++ b/modules/video_output/win32/direct3d11.c
@@ -1823,7 +1823,10 @@ static bool AllocQuadVertices(vout_display_t *vd, d3d_quad_t *quad, video_projec
         quad->indexCount = nbLatBands * nbLonBands * 2 * 3;
     }
     else
+    {
+        msg_Warn(vd, "Projection mode %s not handled", projection);
         return false;
+    }
 
     D3D11_BUFFER_DESC bd;
     memset(&bd, 0, sizeof(bd));
-- 
2.10.1.windows.1

