From 16b6931b6e587c8c9a4a1a5b1cf2344cc06fb936 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@videolabs.io>
Date: Fri, 6 Jan 2017 14:13:56 +0100
Subject: [PATCH 13/39] direct3d11: make sure ppColorspace is always set a
 value

---
 modules/video_output/win32/direct3d11.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/video_output/win32/direct3d11.c b/modules/video_output/win32/direct3d11.c
index d5b926b..82cff6e 100644
--- a/modules/video_output/win32/direct3d11.c
+++ b/modules/video_output/win32/direct3d11.c
@@ -1937,6 +1937,7 @@ static int AllocQuad(vout_display_t *vd, const video_format_t *fmt, d3d_quad_t *
         case COLOR_SPACE_BT601:
             ppColorspace = COLORSPACE_BT601_TO_FULL;
             break;
+        default:
         case COLOR_SPACE_UNDEF:
             if( fmt->i_height > 576 )
                 ppColorspace = COLORSPACE_BT709_TO_FULL;
-- 
2.10.1.windows.1

