From ee89799901e2567743f26055839d25599d0e270c Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Tue, 20 Feb 2018 11:24:43 +0100
Subject: [PATCH 13/13] contrib:mpg123: don't use forbidden APIs on Phone apps

---
 contrib/src/mpg123/rules.mak | 1 +
 1 file changed, 1 insertion(+)

diff --git a/contrib/src/mpg123/rules.mak b/contrib/src/mpg123/rules.mak
index 2d43c8c..278e2eb 100644
--- a/contrib/src/mpg123/rules.mak
+++ b/contrib/src/mpg123/rules.mak
@@ -43,6 +43,7 @@ ifdef HAVE_WIN32
 endif
 ifdef HAVE_WINSTORE
 	$(APPLY) $(SRC)/mpg123/winstore.patch
+	$(APPLY) $(SRC)/mpg123/mpg123-winstore.patch
 endif
 	$(MOVE)
 
-- 
2.10.1.windows.1

