From cf91ff601df2d68ffd75dc9951023946d8232951 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Mon, 19 Feb 2018 15:24:41 +0100
Subject: [PATCH 15/20] d3d11va: try to decode to P010 with 10 bits decoders

---
 modules/codec/avcodec/d3d11va.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/modules/codec/avcodec/d3d11va.c b/modules/codec/avcodec/d3d11va.c
index 0ffb06a..bf6dcf5 100644
--- a/modules/codec/avcodec/d3d11va.c
+++ b/modules/codec/avcodec/d3d11va.c
@@ -530,6 +530,7 @@ static int DxGetInputList(vlc_va_t *va, input_list_t *p_list)
 }
 
 extern const GUID DXVA_ModeHEVC_VLD_Main10;
+extern const GUID DXVA_ModeVP9_VLD_10bit_Profile2;
 static bool CanUseIntelHEVC(vlc_va_t *va)
 {
     vlc_va_sys_t *sys = va->sys;
@@ -569,6 +570,8 @@ static int DxSetupOutput(vlc_va_t *va, const GUID *input, const video_format_t *
     int idx = 0;
     if ( sys->render != DXGI_FORMAT_UNKNOWN )
         processorInput[idx++] = sys->render;
+    if (IsEqualGUID(input, &DXVA_ModeHEVC_VLD_Main10) || IsEqualGUID(input, &DXVA_ModeVP9_VLD_10bit_Profile2))
+        processorInput[idx++] = DXGI_FORMAT_P010;
     processorInput[idx++] = DXGI_FORMAT_NV12;
     processorInput[idx++] = DXGI_FORMAT_420_OPAQUE;
     processorInput[idx++] = DXGI_FORMAT_UNKNOWN;
-- 
2.10.1.windows.1

