From 6967a6ae256bcdd3d2aefc7f778bdbbee99716e1 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Thu, 29 Mar 2018 13:09:59 +0200
Subject: [PATCH 16/38] sap: use the same declaration signature as the
 definition

---
 src/stream_output/sap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/stream_output/sap.c b/src/stream_output/sap.c
index ee5312eb7d..2b5b0e13a2 100644
--- a/src/stream_output/sap.c
+++ b/src/stream_output/sap.c
@@ -79,7 +79,7 @@ static vlc_mutex_t sap_mutex = VLC_STATIC_MUTEX;
 #define MIN_INTERVAL 2
 #define MAX_INTERVAL 300
 
-static void *RunThread (void *);
+noreturn static void *RunThread (void *);
 
 static sap_address_t *AddressCreate (vlc_object_t *obj, const char *group)
 {
-- 
2.19.2

