From 180b0b3580ee7c95051545c57de761c730c0b51b Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@videolabs.io>
Date: Fri, 13 Jan 2017 09:57:52 +0100
Subject: [PATCH 18/39] direct3d11: TEXCOORD0 should be a float4

---
 modules/video_output/win32/direct3d11.c | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/modules/video_output/win32/direct3d11.c b/modules/video_output/win32/direct3d11.c
index e478e74..cb59794 100644
--- a/modules/video_output/win32/direct3d11.c
+++ b/modules/video_output/win32/direct3d11.c
@@ -178,13 +178,13 @@ static const char* globVertexShaderFlat = "\
   struct VS_INPUT\
   {\
     float4 Position   : POSITION;\
-    float2 Texture    : TEXCOORD0;\
+    float4 Texture    : TEXCOORD0;\
   };\
   \
   struct VS_OUTPUT\
   {\
     float4 Position   : SV_POSITION;\
-    float2 Texture    : TEXCOORD0;\
+    float4 Texture    : TEXCOORD0;\
   };\
   \
   VS_OUTPUT VS( VS_INPUT In )\
@@ -205,13 +205,13 @@ static const char* globVertexShaderProjection = "\
   struct VS_INPUT\
   {\
     float4 Position   : POSITION;\
-    float2 Texture    : TEXCOORD0;\
+    float4 Texture    : TEXCOORD0;\
   };\
   \
   struct VS_OUTPUT\
   {\
     float4 Position   : SV_POSITION;\
-    float2 Texture    : TEXCOORD0;\
+    float4 Texture    : TEXCOORD0;\
   };\
   \
   VS_OUTPUT VS( VS_INPUT In )\
@@ -241,7 +241,7 @@ static const char* globPixelShaderDefault = "\
   struct PS_INPUT\
   {\
     float4 Position   : SV_POSITION;\
-    float2 Texture    : TEXCOORD0;\
+    float4 Texture    : TEXCOORD0;\
   };\
   \
   float4 PS( PS_INPUT In ) : SV_TARGET\
@@ -254,6 +254,7 @@ static const char* globPixelShaderDefault = "\
   }\
 ";
 
+/* for NV12/P010 source */
 static const char *globPixelShaderBiplanarYUV_2RGB = "\
   cbuffer PS_CONSTANT_BUFFER : register(b0)\
   {\
@@ -275,7 +276,7 @@ static const char *globPixelShaderBiplanarYUV_2RGB = "\
   struct PS_INPUT\
   {\
     float4 Position   : SV_POSITION;\
-    float2 Texture    : TEXCOORD0;\
+    float4 Texture    : TEXCOORD0;\
   };\
   \
   float4 PS( PS_INPUT In ) : SV_TARGET\
@@ -313,7 +314,7 @@ static const char *globPixelShaderBiplanarYUYV_2RGB = "\
   struct PS_INPUT\
   {\
     float4 Position   : SV_POSITION;\
-    float2 Texture    : TEXCOORD0;\
+    float4 Texture    : TEXCOORD0;\
   };\
   \
   float4 PS( PS_INPUT In ) : SV_TARGET\
-- 
2.10.1.windows.1

