From 905fa3d4a9268db8f8e5935832a7d1160da48cc5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo@beauzee.fr>
Date: Mon, 9 Apr 2018 13:57:01 +0200
Subject: [PATCH 36/37] archive: Link with -lbcrypt on windows

---
 modules/stream_extractor/Makefile.am | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/modules/stream_extractor/Makefile.am b/modules/stream_extractor/Makefile.am
index 6b37a01bbd..be5a52e4d9 100644
--- a/modules/stream_extractor/Makefile.am
+++ b/modules/stream_extractor/Makefile.am
@@ -5,5 +5,8 @@ libarchive_plugin_la_SOURCES = stream_extractor/archive.c
 libarchive_plugin_la_CFLAGS = $(AM_CFLAGS) $(ARCHIVE_CFLAGS)
 libarchive_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(stream_extractordir)'
 libarchive_plugin_la_LIBADD = $(ARCHIVE_LIBS)
+if HAVE_WIN32
+libarchive_plugin_la_LIBADD += -lbcrypt
+endif
 EXTRA_LTLIBRARIES += libarchive_plugin.la
 stream_extractor_LTLIBRARIES += $(LTLIBarchive)
-- 
2.11.0

