From ed64706e4ef4ee07d48a5895326538d66cf5d3f8 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@videolabs.io>
Date: Wed, 25 Jan 2017 10:45:50 +0100
Subject: [PATCH 45/53] avcodec: simplify the log generation

---
 modules/codec/avcodec/video.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/modules/codec/avcodec/video.c b/modules/codec/avcodec/video.c
index 9086c20..dc477f8 100644
--- a/modules/codec/avcodec/video.c
+++ b/modules/codec/avcodec/video.c
@@ -1173,16 +1173,14 @@ static int lavc_dr_GetFrame(struct AVCodecContext *ctx, AVFrame *frame,
         if (pic->p[i].i_pitch % aligns[i])
         {
             if (!atomic_exchange(&sys->b_dr_failure, true))
-                msg_Warn(dec, "plane %d: pitch not aligned (%d%%%d): %s",
-                         i, pic->p[i].i_pitch, aligns[i],
-                         "disabling direct rendering");
+                msg_Warn(dec, "plane %d: pitch not aligned (%d%%%d): disabling direct rendering",
+                         i, pic->p[i].i_pitch, aligns[i]);
             goto error;
         }
         if (((uintptr_t)pic->p[i].p_pixels) % aligns[i])
         {
             if (!atomic_exchange(&sys->b_dr_failure, true))
-                msg_Warn(dec, "plane %d not aligned: %s", i,
-                         "disabling direct rendering");
+                msg_Warn(dec, "plane %d not aligned: disabling direct rendering", i);
             goto error;
         }
     }
-- 
2.10.1.windows.1

