From a441ef4dd84543a33816911f37d5705516cd8432 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@videolabs.io>
Date: Tue, 31 Jan 2017 12:36:22 +0100
Subject: [PATCH 47/55] direct3d11: no need to recheck the picQuadConfig again

---
 modules/video_output/win32/direct3d11.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/modules/video_output/win32/direct3d11.c b/modules/video_output/win32/direct3d11.c
index 8241bc34f3..f9d6589576 100644
--- a/modules/video_output/win32/direct3d11.c
+++ b/modules/video_output/win32/direct3d11.c
@@ -775,9 +775,6 @@ static picture_pool_t *Pool(vout_display_t *vd, unsigned pool_size)
     ID3D11Texture2D  *textures[pool_size * D3D11_MAX_SHADER_VIEW];
     HRESULT           hr;
 
-    const d3d_format_t *cfg = GetOutputFormat(vd, vd->fmt.i_chroma, 0, true, false);
-    assert(cfg == vd->sys->picQuadConfig);
-
     ID3D10Multithread *pMultithread;
     hr = ID3D11Device_QueryInterface( sys->d3ddevice, &IID_ID3D10Multithread, (void **)&pMultithread);
     if (SUCCEEDED(hr)) {
-- 
2.11.0

