From 49da38cb17f983267604ced6b14d001f86b6050c Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@videolabs.io>
Date: Tue, 31 Jan 2017 15:13:34 +0100
Subject: [PATCH 57/62] direct3d11: fix green line in software decoding

---
 modules/video_output/win32/direct3d11.c | 18 ++++++++----------
 1 file changed, 8 insertions(+), 10 deletions(-)

diff --git a/modules/video_output/win32/direct3d11.c b/modules/video_output/win32/direct3d11.c
index 45e5681..01fe2ae 100644
--- a/modules/video_output/win32/direct3d11.c
+++ b/modules/video_output/win32/direct3d11.c
@@ -835,7 +835,7 @@ static picture_pool_t *Pool(vout_display_t *vd, unsigned pool_size)
     }
 
 #ifdef HAVE_ID3D11VIDEODECODER
-    if (is_d3d11_opaque(vd->fmt.i_chroma) && sys->legacy_shader)
+    if (!is_d3d11_opaque(vd->fmt.i_chroma) || sys->legacy_shader)
     {
         /* we need a staging texture */
         video_format_t staging_fmt;
@@ -1193,14 +1193,16 @@ static void Prepare(vout_display_t *vd, picture_t *picture, subpicture_t *subpic
 {
     vout_display_sys_t *sys = vd->sys;
 
-#ifdef HAVE_ID3D11VIDEODECODER
-    if (is_d3d11_opaque(picture->format.i_chroma) && sys->legacy_shader) {
+    if (!is_d3d11_opaque(picture->format.i_chroma) || sys->legacy_shader) {
         picture_sys_t *p_sys = picture->p_sys;
         D3D11_TEXTURE2D_DESC texDesc;
         if( sys->context_lock != INVALID_HANDLE_VALUE )
         {
             WaitForSingleObjectEx( sys->context_lock, INFINITE, FALSE );
         }
+        if ( !is_d3d11_opaque(picture->format.i_chroma))
+            Direct3D11UnmapPoolTexture(picture);
+
         ID3D11Texture2D_GetDesc(sys->stagingSys.texture[0], &texDesc);
         D3D11_BOX box = {
             .bottom = texDesc.Height,
@@ -1215,7 +1217,6 @@ static void Prepare(vout_display_t *vd, picture_t *picture, subpicture_t *subpic
         if ( sys->context_lock != INVALID_HANDLE_VALUE)
             ReleaseMutex( sys->context_lock );
     }
-#endif
 
     if (subpicture) {
         int subpicture_region_count    = 0;
@@ -1271,15 +1272,10 @@ static void Display(vout_display_t *vd, picture_t *picture, subpicture_t *subpic
 
     ID3D11DeviceContext_ClearDepthStencilView(sys->d3dcontext, sys->d3ddepthStencilView, D3D11_CLEAR_DEPTH | D3D11_CLEAR_STENCIL, 1.0f, 0);
 
-    if ( !is_d3d11_opaque(picture->format.i_chroma))
-        Direct3D11UnmapPoolTexture(picture);
-
     /* Render the quad */
-#ifdef HAVE_ID3D11VIDEODECODER
-    if (is_d3d11_opaque(picture->format.i_chroma) && sys->legacy_shader)
+    if (!is_d3d11_opaque(picture->format.i_chroma) || sys->legacy_shader)
         DisplayD3DPicture(sys, &sys->picQuad, sys->stagingSys.resourceView);
     else
-#endif
         DisplayD3DPicture(sys, &sys->picQuad, picture->p_sys->resourceView);
 
     if (subpicture) {
@@ -1861,6 +1857,8 @@ static void SetupQuadFlat(d3d_vertex_t *dst_data, const video_format_t *fmt, WOR
     float top    =  (float) (2*fmt->i_y_offset + fmt->i_visible_height) / (float) fmt->i_visible_height;
     float bottom = -(float) (2*fmt->i_height-fmt->i_visible_height-2*fmt->i_y_offset) / (float) fmt->i_visible_height;
 
+    /* TODO off by 1 when legacy_shader is not set to avoid a copy */
+
     // bottom left
     dst_data[0].position.x = left;
     dst_data[0].position.y = bottom;
-- 
2.10.1.windows.1

