From 500d30c7e6e7afb84b2ba1634346d87e6dd6e3f4 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@videolabs.io>
Date: Wed, 8 Jun 2016 17:40:34 +0200
Subject: [PATCH] avcodec: enable thread hardware decoding with FFmpeg on
 Windows

---
 modules/codec/avcodec/video.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/codec/avcodec/video.c b/modules/codec/avcodec/video.c
index a97fec2..130148c 100644
--- a/modules/codec/avcodec/video.c
+++ b/modules/codec/avcodec/video.c
@@ -1233,7 +1233,7 @@ static enum PixelFormat ffmpeg_GetFormat( AVCodecContext *p_context,
     if (!can_hwaccel)
         return swfmt;
 
-#if (LIBAVCODEC_VERSION_MICRO >= 100) /* FFmpeg only */
+#if !defined(_WIN32) && (LIBAVCODEC_VERSION_MICRO >= 100) /* FFmpeg only */
     if (p_context->active_thread_type)
     {
         msg_Warn(p_dec, "thread type %d: disabling hardware acceleration",
